packages/perspective-viewer-d3fc/src/js/zoom/zoomableChart.js

```diff
@@ -37,12 +38,27 @@ export default () => {
 
                 const noZoom = transform.k === 1 && transform.x === 0 && transform.y === 0;
 
-                getZoomControls(selection)
-                    .style("display", noZoom ? "none" : "")
-                    .select("#zoom-reset")
-                    .on("click", () => {
-                        selection.select(chartPlotArea).call(zoom.transform, d3.zoomIdentity);
+                const zoomControls = getZoomControls(selection).style("display", noZoom ? "none" : "");
+                zoomControls.select("#zoom-reset").on("click", () => selection.select(chartPlotArea).call(zoom.transform, d3.zoomIdentity));
+
+                const oneYear = zoomControls.select("#one-year").style("display", dateAxis ? "" : "none");
+                if (dateAxis) {
+                    const xRange = xCopy.range();

```

Comment:
Looks like the `xRange` variable might as well be defined inside the `click` handler.

Reply:
I think the only reason I hadn't done that is because it's only used for the date range button. I'll move it.

Reply:
> Looks like the `xRange` variable might as well be defined inside the `click` handler.

Only just realised that I misunderstood this!